### PR TITLE
server/statistics: add abnormal peer duration metric

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

* Add abnormal peer duration metric.

If a region has `down-peer` or `miss-peer` for a long time, we need a metric to trigger alertmanager rule. such as the following rule:

```
histogram_quantile(0.99, sum(rate(pd_regions_abnormal_peer_duration_seconds_bucket[5m])) by (type,le)) > 10 * 60
```

### Check List

Tests

- No code

Code changes

- N/A

Side effects

- Increased code complexity

Related changes

- N/A

### Release note

N/A

### CODE REVIEW

**Comment by crazycs520** [comment]
@HunDunDM @nolouch PTAL

**Comment by crazycs520** [comment]
@nolouch @HunDunDM PTAL

File: server/statistics/metrics.go [hunk 1]
```diff
@@ -127,6 +127,29 @@ var (
...
+	regionAbnormalPeerDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "pd",
+			Subsystem: "regions",
+			Name:      "abnormal_peer_duration_seconds",
+			Help:      "Bucketed histogram of processing time (s) of handled success cmds.",
+			Buckets:   prometheus.ExponentialBuckets(1, 1.4, 30), // 1s ~ 6.72 hours
+		}, []string{"type"})
+
+	regionIsDangerousCounter = prometheus.NewCounter(
```
Comment (nolouch) [inline server/statistics/metrics.go:140]: I think pd cannot know the abnomal region because the region cannot elect the leader and report to PD.

Comment (crazycs520) [reply server/statistics/metrics.go:140]: Great catch.

**Comment by crazycs520** [comment]
@nolouch @HunDunDM PTAL

**Review by nolouch — COMMENTED** [review]
lgtm. could you show the test result.

**Comment by crazycs520** [comment]
> lgtm. could you show the test result.

Total 3 tikv, after I kill one tikv:

image

**Review by nolouch — APPROVED** [review]

**Comment by crazycs520** [comment]
@rleungx PTAL

File: server/core/region.go [hunk 2]
```diff
@@ -436,6 +436,25 @@ func (r *RegionInfo) GetReplicationStatus() *replication_modepb.RegionReplicatio
 	return r.replicationStatus
 }
 
+// GetNormalVotersCount gets the normal voters count.
+// Normal voter means the peer.role is not learner and the peer status not in pending or down.
+func (r *RegionInfo) GetNormalVotersCount() int {
```
Comment (rleungx) [inline server/core/region.go:441]: Where do you use it?

Comment (crazycs520) [reply server/core/region.go]: Sorry, forgot to remove it.
Done.

**Review by rleungx — APPROVED** [review]

**Pull request merged**